Mappers/HttpMapper.php

```diff
@@ -27,128 +36,152 @@
      */
     private $httpMapperConfig = null;
 
+    /**
+     * @var string
+     */
+    protected static $contentType = self::CONTENT_TYPE_JSON;
+
+    /**
+     * HttpMapper constructor.
+     * @param array $httpMapperConfig
+     * @param HttpClient $httpClient
+     */
     public function __construct(array $httpMapperConfig, HttpClient $httpClient)
     {
-        $this->setHttpMapperConfig($httpMapperConfig)->setHttpClient($httpClient);
+        $this->httpClient = $httpClient;
+        $this->httpMapperConfig = $httpMapperConfig;
     }
 
     /**
-     * @param $conditions
-     * @return Collection
-     * @throws EnergonException
+     * @param array $objectData
+     * @return ArraySerializable
+     * @throws HttpResourceNotFoundException
+     * @throws InfrastructureException
+     * @throws ValidationException
+     * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    public function load($conditions)
+    public function create(array $objectData): ArraySerializable
     {
-        return $this->sendHttpRequest('GET', $this->getHttpMapperConfig(self::LOAD_URL), ['query' => $conditions]);
+        return $this->sendHttpRequest('POST', $this->createUrl(), [], $objectData);
     }
 
     /**
-     * @param $identifierValue
-     * @return ArraySerializable
-     * @throws EnergonException
+     * @param array $objectData
+     * @return PaginationCollection|mixed
+     * @throws HttpResourceNotFoundException
+     * @throws InfrastructureException
+     * @throws ValidationException
+     * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    public function get($identifierValue)
+    public function update(array $objectData)
     {
-        return $this->sendHttpRequest('GET', $this->getUrlWithIdentifier($identifierValue, self::GET_URL), []);
+        return $this->sendHttpRequest('PUT', $this->updateUrl($objectData), [], $objectData);
     }
 
     /**
-     * @param $identifierValue
-     * @return bool
-     * @throws EnergonException
+     * @param SearchCriteria $filter
+     * @return PaginationCollection
+     * @throws HttpResourceNotFoundException
+     * @throws InfrastructureException
+     * @throws ValidationException
+     * @throws \GuzzleHttp\Exception\GuzzleException
      */
-    public function delete($identifierValue)
+    public function load(SearchCriteria $filter): PaginationCollection
```

Comment:
Mostly all API requires headers. How this load should work ? Get? Delete?  This will work only for open API's.